tf/tfprocess.py

```diff
@@ -113,16 +141,42 @@ def __init__(self, cfg):
                                      self.cfg['name'])
 
         # Network structure
-        self.RESIDUAL_FILTERS = self.cfg['model']['filters']
-        self.RESIDUAL_BLOCKS = self.cfg['model']['residual_blocks']
-        self.SE_ratio = self.cfg['model']['se_ratio']
+        self.RESIDUAL_FILTERS = self.cfg['model'].get('filters', 0)
+        self.RESIDUAL_BLOCKS = self.cfg['model'].get('residual_blocks', 0)
+        self.SE_ratio = self.cfg['model'].get('se_ratio', 0)
+        self.encoder_layers = self.cfg['model'].get('encoder_layers', 0)
+        self.encoder_heads = self.cfg['model'].get('encoder_heads', 2)
+        assert(bool(self.RESIDUAL_BLOCKS > 0) != bool(self.encoder_layers > 0),
+                "Nets with both encoder layers and residual blocks are not supported")
+        if self.encoder_layers > 0:
+            self.RESIDUAL_FILTERS = self.cfg['model']['embedding_size']
+            self.embedding_size = self.RESIDUAL_FILTERS
+            
         self.policy_channels = self.cfg['model'].get('policy_channels', 32)
         self.pol_embedding_size = self.cfg['model'].get('pol_embedding_size', self.RESIDUAL_FILTERS)
+        self.val_embedding_size = self.cfg['model'].get('value_embedding_size', 32)
+        self.mov_embedding_size = self.cfg['model'].get('moves_left_embedding_size', 8)
+        #policy head
         self.pol_encoder_layers = self.cfg['model'].get('pol_encoder_layers', 1)
-        self.pol_encoder_heads = self.cfg['model'].get('pol_encoder_heads', 2)
+        self.pol_encoder_heads = self.encoder_heads
```

Comment:
I think this should be self.cfg['model'].get('pol_encoder_heads', self.encoder_heads) ?